cl/corpus_importer/management/commands/make_aws_manifest_from_search.py

```diff
@@ -0,0 +1,186 @@
+
+
+def build_base_query(options: dict[str, Any]) -> Search:
+    """
+    Constructs a base Full text Elasticsearch query using the provided options.
+
+    Args:
+        options (dict[str, Any]): A dictionary containing search options:
+            - search_text (str): The text to search for.
+            - search_type (str): The type of search.
+            - filed_after (date | str): Filter documents filed **on or after**
+            the specified date.
+            - filed_before (date | str): Filter documents filed **on or before**
+            the specified date.
+            - es_page_size (int): The desired number of results per page.
+
+    Returns:
+        Search: The constructed Elasticsearch query object.
+
+    Raises:
+        NotImplementedError: If the provided search_type is not supported.
+    """
+    cd = {
+        "q": options["search_text"],
+        "type": options["search_type"],
+    }
+    match options["search_type"]:
+        case SEARCH_TYPES.OPINION:
+            unique_sorting_key = "id"
+            cd["order_by"] = "score desc"
+            search_query = OpinionDocument.search()
+            search_fields = SEARCH_OPINION_QUERY_FIELDS.copy()
+            search_fields.extend(["type", "text", "caseName", "docketNumber"])
+            q_should = build_fulltext_query(
+                search_fields, cd["q"], only_queries=True
+            )
+            filters = [Q("match", cluster_child="opinion")]
+            if options["filed_after"] or options["filed_before"]:
+                filters.extend(
+                    build_daterange_query(
+                        "dateFiled",
+                        options.get("filed_before", ""),
+                        options.get("filed_after", ""),
+                    )
+                )
+            query = Q("bool", should=q_should, minimum_should_match=1)
+            query.filter = reduce(operator.iand, filters)
+        case _:
+            raise NotImplemented
+
+    search_query = search_query.query(query)
+    # Limit the search response to only include the "id" field for each
+    # document. This reduces the amount of data transferred and improves
+    # performance.
+    search_query = search_query.source(includes=["id"])
+    # Set the number of results to return per page (`options["es_page_size"]`).
+    # We add 1 to this value to enable pagination detection. This extra result
+    # allows us to determine if there are more pages available after the
+    # current set of results.
+    search_query = search_query.extra(size=options["es_page_size"] + 1)
+    # Build the sorting settings for an ES query to work with the
+    # 'search_after' pagination
+    default_sorting = build_sort_results(cd, False, "v4")
+    unique_sorting = build_sort_results(
+        {"type": cd["type"], "order_by": f"{unique_sorting_key} desc"},
+        False,
+        "v4",
+    )
+    search_query = search_query.sort(default_sorting, unique_sorting)
```

Comment:
Following my previous comment, we can remove the `default_sorting` (based on score) and keep only the `unique_sorting`, which is required for the `search_after` pagination.

Reply:
perfect! I'll update the sorting.